sonaranalyzer-dotnet/src/SonarAnalyzer.CSharp/Metrics/Metrics.cs

```diff
@@ -146,6 +147,13 @@ public override int GetComplexity(SyntaxNode node)
             return walker.CyclomaticComplexity;
         }
 
+        public override int GetCognitiveComplexity(SyntaxNode node)
+        {
+            var walker = new CognitiveComplexityWalker();
+            walker.Visit(node);

```

Comment:
In order to prevent exceptions when generating the metrics (and thus breaking all metrics) we return -1 for cognitive if the visit contained errors. Having value -1 will make the metrics imported in the java part not report the metric to SQ.